# Sintaxe de Python

> Antes de iniciar qualquer notebook, é importante que você limpe as saídas de todas as células para que você só veja o resultado de uma célula quando executá-la. Para isso, utilize o menu **Cell > All Output > Clear **.

## Variáveis

Para criar as mais diversas aplicações, precisamos armazenar e manipular diferentes tipos de dados. Fazemos usando **variáveis**. Uma **variável** armazena um item de dado (valor) e recebe um nome específico.

Por exemplo:

In [ ]:
num = 9

A linha acima cria uma variável ```num``` que armazena o número ```9```.

Na célula de código abaixo, crie uma variável chamada ```contador``` que armazena o número ```7```.


Não se esqueça de executar a célula acima (pressionando ```Shift + Enter```).

Ótimo! Você acabou de armazenar um número inteiro em uma variável. Números são um dos tipos de dados que usamos em programação. Outro tipo que também usamos são booleanos.

## Booleanos

Um booleano é como um interruptor de luz, que só pode estar em dois estados: **ligado** ou **desligado**. Assim, um booleano pode possuir apenas dois valores: **True** (verdadeiro) ou **False** (falso).

Você pode usar variáveis para armazenar booleanos da seguinte maneira:

```python
a = True
b = False
```

Na célula abaixo, crie uma variável booleana ``` aprendendo_python``` e atribua o valor ```True```. Sempre se lembre de executar a célula com ```Shift+Enter```.

## Números Inteiros

Agora você já sabe como usar variáveis para armazenar valores booleanos. Variáveis numéricas são criadas de modo semelhante:

```python
semestre = 3
```
Na célula abaixo, crie uma variável ```idade``` e armazene um número correspondendo à sua idade.

Suponha que você criou a variável idade com o valor 20, ou seja, ```idade = 20```. Você pode alterar o valor de uma variável, atribuindo um outro valor à idade:

```
idade = 19

```

Tente você também. Na célula abaixo foi criada uma variável tamanho com valor 10.

In [9]:
tamanho = 10

 Atribua o novo valor 15 a tamanho na linha abaixo:

In [ ]:
tamanho = 

Abaixo está um código que mostra o valor de tamanho. Ao executá-la, é mostrado o valor 15?

In [11]:
print(tamanho)

10


## Números reais

Você também pode armazenar números não inteiros em variáveis. Devido a uma limitação de representação nem todos os números reais podem ser representados em um computador, mas uma boa parte. 

Para isso, usamos o ```.``` (ponto) como separador de casas decimais. Por exemplo:

```python
altura = 1.78
```

Na célula abaixo, crie uma variável altura e atribua o valor correspondente à sua altura.

## Espaços

Em Python, espaços são usados para estruturar o código. Espaços são importantes e, portanto, você deve tomar cuidado como você os utiliza dentro do código.

Por exemplo, a célula abaixo está mal formatada. Existe um *espaço inadequado* no início da segunda linha. Tente executá-la e veja o que acontece:

In [1]:
tamanho = 15
 idade = 19

IndentationError: unexpected indent (<ipython-input-1-6d2a838c2a0b>, line 2)

Você verá algo como:
```
File "<ipython-input-1-6d2a838c2a0b>", line 2
    idade = 19
    ^
IndentationError: unexpected indent
```

Para corrigir o erro acima, basta remover o espaço no início da linha onde está variável idade e re-executar a célula.

Veja que o interpretador Python executa cada célula, linha por linha, e verifica se existe algum erro.

## Comentários

### Comentários de uma linha só

Você já viu que em algumas células usamos o símbolo **#** para criar comentários. Um comentário é uma linha de texto que o Python não executará como código. Comentários existem apenas para serem lidos por humanos.

Comentários tornam o seu programa mais fácil de ser lido e entendido. Quando você revisitar os códigos escritos no passado ou então compartilhar o seu código com colaboradores, eles podem ler os comentários e facilmente compreender o que o código faz.

Na primeira linha da célula abaixo crie um comentário que explica pra que serve a variável ```misterio```. 

> Observação: Evite o uso de acentos e ç em nomes de variáveis.

In [2]:

misterio = 3.1416

> Observação: Note o uso do ponto como separador de casas decimais. Em Python **não usamos a vírgula, e sim o ponto**.

### Comentários de múltiplas linhas

O símbolo **#** irá comentar apenas uma única linha de texto. Mesmo podendo criar um comentário com múltiplas linhas, iniciando cada linha com **#**, como foi feito anteriormente, existe uma maneira mais elegante. Para criar comentários grandes, com várias linhas de texto, você pode demarcar todo o bloco de texto do comentário por três aspas duplas (```"""```) no início e três aspas duplas no final (```"""```):

In [ ]:
""" Esse é um comentário com
várias 
linhas.
Não preciso iniciá-las com #
"""

## Operações Matemáticas

Python também permite operações matemáticas. Você pode somar, subtrair, multiplicar e dividir números da seguinte maneira:

In [4]:
soma = 42 + 27
subtracao = 103 - 254
multiplicacao = 100 * 0.5
divisao = 100 / 9

Na célula abaixo, crie uma variável ```horas``` que armazena o número total de horas no mês de janeiro.

In [5]:
horas = 
print(horas)

SyntaxError: invalid syntax (<ipython-input-5-7a99a3c0f823>, line 1)

### Exponenciação

Além das operações simples, também podemos elevar um número a outro em Python. Por exemplo, para calcular $3^2$, fazemos:

In [6]:
nove = 3 ** 2
print(nove)

9


Na célula acima, criamos uma variável chamada nove e atribuímos ```3 ** 2```. Note o uso do operador ```**``` para denotar exponenciação.  

Na célula abaixo, crie uma variável oito e armazene o valor $2^3$.

In [7]:
oito = 
print(oito)

SyntaxError: invalid syntax (<ipython-input-7-2d37d570ec12>, line 1)

## Operadores Divisao Inteira e Modulo

Sempre que quisermos a parte inteira da divisão entre dois números, usamos o operador ```//```. E sempre que quisermos o resto da divisão inteira, usamos o operador modulo: ```%```.

Por exemplo, a célula abaixo não faz a divisão inteira.

In [10]:
divisao = 5 / 2
print(divisao)

2.5


Para calcular a divisão inteira e o resto, fazemos:

In [11]:
divisao = 5 // 2
resto = 5 % 2
print(divisao)
print(resto)

2
1


Pratique o uso desses operadores, mostrando na célula abaixo o total de dias existentes em 80 horas e quantas horas sobram.

In [ ]:
dias = 
horas = 
print(dias)
print(horas)

## Precedência de operadores

Quando múltiplos operadores aparecerem em uma mesma expressão, devemos resolvê-los de acordo com a prioridade mostrada na figura abaixo:

![Precedência de operadores aritméticos](images/operadores_aritmeticos_precedencia.png)

## Revisão
Muito bem, até agora aprendemos sobre:

* **Variáveis**, que armazenam valores para serem utilizados posteriormente
* **Tipos de dados**, tais como números e booleandos
* **Espaços**, usados para delimitar blocos em Python e precisam ser respeitados
* **Comentários**, usados para deixar o código mais legível
* **Operações aritméticas**, incluindo +, -, *, /, // e %

### Exercícios
Na célula de código abaixo, faça o seguinte:

1. Escreva um comentário na primeira linha (pode ser qualquer coisa)
2. Crie uma variável ```feliz``` com o valor ```True```
3. Crie uma outra variável ```lado``` com o valor 2.5
4. Crie uma variável ```area``` com valor de ```lado``` ao quadrado

Ao final execute a sua célula e a célula abaixo. Se nenhuma mensagem de erro aparecer, você concluiu essa atividade com sucesso!

Ao terminar, você já pode ir para o próximo notebook: [Strings e saída de dados](03-strings.ipynb).